src/routes/messages/entities/message.entity.ts

```diff
@@ -7,6 +12,53 @@ export enum MessageStatus {
   Confirmed = 'CONFIRMED',
 }
 
+class TypedDataDomain {
+  @ApiPropertyOptional({ nullable: true }) name?: string;
+  @ApiPropertyOptional({ nullable: true }) version?: string;
+  @ApiPropertyOptional({ nullable: true }) chainId?: unknown;
+  @ApiPropertyOptional({ nullable: true }) verifyingContract?: string;
+  @ApiPropertyOptional({ nullable: true }) salt?: string | Array<number>;
+}
+
+class TypedDataTypes {
+  @ApiProperty() name: string;
+  @ApiProperty() type: string;
+
+  constructor(name: string, type: string) {
+    this.name = name;
+    this.type = type;
+  }
+}
+
+type TypedMessageTypes = Record<string, Array<TypedDataTypes>>;
+
+class EIP712TypedData {
+  @ApiProperty({ type: () => TypedDataDomain })
+  domain: TypedDataDomain;
+
+  @ApiProperty({
+    type: 'object',
+    additionalProperties: {
+      type: 'array',
+      items: { $ref: getSchemaPath(TypedDataTypes) },
+    },
+  })
+  types: TypedMessageTypes;
+  @ApiProperty({ type: 'object', additionalProperties: true })
+  message: Record<string, unknown>;
+
+  constructor(
+    domain: TypedDataDomain,
+    types: Record<string, Array<TypedDataTypes>>,
+    message: Record<string, unknown>,
+  ) {
+    this.domain = domain;
+    this.types = types;
+    this.message = message;
+  }
+}
```

Comment:
Nit: I'd move these to their own entity file (can be in this folder) as they are not necessarily `Message`-related.